### PR TITLE
Do not allow blank email addresses to slip through validation.

### PR DESCRIPTION
**Pull request by alxdarksage**
I found while testing through the BSM that I could create a user with a blank email address and a phone number, that triggered a 500 error (we try and send an email to the blank email address).

Fixed with stronger validation.

### CODE REVIEW

File: app/org/sagebionetworks/bridge/validators/IdentifierUpdateValidator.java [hunk 1]
```diff
@@ -44,9 +46,15 @@ public void validate(Object object, Errors errors) {
         int updateFields = 0;
         if (update.getPhoneUpdate() != null) {
             updateFields++;
+            if (!Phone.isValid(update.getPhoneUpdate())) {
+                errors.rejectValue("phoneUpdate", "does not appear to be a phone number");
+            }
```
Comment (DwayneJengSage) [inline app/org/sagebionetworks/bridge/validators/IdentifierUpdateValidator.java:51]: Should we add tests to IdentifierUpdateValidatorTest to make sure we catch invalid emails and phone numbers?

File: app/org/sagebionetworks/bridge/validators/StudyParticipantValidator.java [hunk 2]
```diff
@@ -51,7 +50,7 @@ public void validate(Object object, Errors errors) {
                 errors.rejectValue("phone", "does not appear to be a phone number");
             }
             // If provided, email must be valid
-            if (isNotBlank(email) && !EMAIL_VALIDATOR.isValid(email)) {
+            if (email != null && !EMAIL_VALIDATOR.isValid(email)) {
                 errors.rejectValue("email", "does not appear to be an email address");
             }
```
Comment (DwayneJengSage) [inline app/org/sagebionetworks/bridge/validators/StudyParticipantValidator.java:55]: If we're treating blank emails as invalid emails rather than as no email specified, we should probably make the change further up in the "email, phone, or externalId is required" section. Either null out blank emails, or check email != null instead if isBlank(email).

File: test/org/sagebionetworks/bridge/validators/StudyParticipantValidatorTest.java [hunk 3]
```diff
@@ -126,6 +126,18 @@ public void emailPhoneOrExternalIdRequired() {
         assertValidatorMessage(validator, withEmail(null), "StudyParticipant", "email, phone, or externalId is required");
     }
     
+    @Test
+    public void emailCannotBeBlank() {
+        validator = new StudyParticipantValidator(externalIdService, study, true);
+        assertValidatorMessage(validator, withEmail(""), "StudyParticipant", "email, phone, or externalId is required");
+    }
```
Comment (DwayneJengSage) [inline test/org/sagebionetworks/bridge/validators/StudyParticipantValidatorTest.java]: Also, to ensure EMAIL_VALIDATOR catches it, we should test both empty strings and strings of all whitespace.

**Review by DwayneJengSage — CHANGES_REQUESTED** [review]

File: app/org/sagebionetworks/bridge/validators/IdentifierUpdateValidator.java (hunk 1 above)
Comment (alxdarksage) [reply app/org/sagebionetworks/bridge/validators/IdentifierUpdateValidator.java:51]: Added test of email validation (neither empty nor junk value pass). Invalid phone test exists.

File: app/org/sagebionetworks/bridge/validators/StudyParticipantValidator.java (hunk 2 above)
Comment (alxdarksage) [reply app/org/sagebionetworks/bridge/validators/StudyParticipantValidator.java:55]: Makes sense, changed this to null pointer check only. Blanks will receive a different message (invalid email) instead.

File: test/org/sagebionetworks/bridge/validators/StudyParticipantValidatorTest.java (hunk 3 above)
Comment (alxdarksage) [reply test/org/sagebionetworks/bridge/validators/StudyParticipantValidatorTest.java]: Added

**Review by DwayneJengSage — APPROVED** [review]
